activesupport/lib/active_support/xml_mini.rb

```diff
@@ -71,7 +71,7 @@ def content_type
             begin
               BigDecimal(number)
             rescue ArgumentError
-              BigDecimal("0")
+              BigDecimal(number.to_f.to_s)
```

Comment:
Why not `BigDecimal(0)`?

Reply:
`BigDecimal("0")` returns the same value as `BigDecimal(0)`: `0.0`. If this is what we want, we can tweak the test.

I understand from the [test](https://github.com/rails/rails/blob/master/activesupport/test/xml_mini_test.rb#L285) that we expect `BigDecimal("123,003")` to return `123.0`, and to do that we need to pass it a cleaned-up string.

Reply:
Ok. I got it. Before `BigDecimal("123,003")` did not raise and now it does.

Reply:
Exactly.
Thank you.